### DISCUSSION TITLE
Moving editor-only methods to their own files

### DISCUSSION THREAD

**Discussion by KoBeWi** [godotengine/godot-proposals#9208] [General Discussions]
We have many methods that are editor only and they are mixed with regular Node methods and wrapped in ifdefs. Like there is whole block in CanvasItem:
https://github.com/godotengine/godot/blob/7d2ca2d8ac49cde9767e00b70f9eaf1920eb266d/scene/main/canvas_item.cpp#L1078-L1095
There are also methods, which are used only in the editor, but never got wrapped (see https://github.com/godotengine/godot/pull/86743).

We are already separating compatibility methods with `compat.inc` files. What if we did the same for editor? E.g. in the above case there would be `canvas_item.editor.cpp` with `_bind_editor_methods()`. It would make make these methods better separated. The same can be applied to all non-editor files that have TOOLS_ENABLED stuff.

Putting aside how much work it would require, would it be beneficial to the codebase?

**Comment by AThousandShips** [comment]
I'd call it `tool` instead as it's not entirely editor restricted, and to match the define
